core/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -499,10 +497,10 @@ public ActionRequestValidationException validate() {
         }
         for (ActionRequest<?> request : requests) {
             // We first check if refresh has been set
-            if ((request instanceof DeleteRequest && ((DeleteRequest)request).refresh()) ||
-                    (request instanceof UpdateRequest && ((UpdateRequest)request).refresh()) ||
-                    (request instanceof IndexRequest && ((IndexRequest)request).refresh())) {
-                    validationException = addValidationError("Refresh is not supported on an item request, set the refresh flag on the BulkRequest instead.", validationException);
+            if (request instanceof WriteRequest<?>
```

Comment:
do we expect anything else then WriteRequest here?  maybe just improve typing on the requests field?